.github/workflows/integrate.yaml

```diff
@@ -8,15 +8,39 @@ on:
 
 jobs:
   quality:
-    name: Quality
+    name: 🔍 Quality
+
     runs-on: ubuntu-latest
+
     steps:
-      - name: Checkout
+      - name: 📥 Checkout
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
-      - name: Setup Biome
+        with:
+          # Disable credentials
+          persist-credentials: false
+
+      - name: 🔧 Setup Biome
         uses: biomejs/setup-biome@c016c38f26f2c4a6eb3662679143614a254263fd # v2
-      - name: Run Biome checks
+
+      - name: 🏃 Run Biome checks
         run: biome ci --reporter=github
-  package:
-    name: Package
-    uses: ./.github/workflows/~package-stable.yaml
+
+  build:
+    name: Build
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: 📥 Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+        with:
+          # Disable credentials
+          persist-credentials: false
+
+      - name: 🛠️ Setup PNPM
+        uses: pnpm/action-setup@v4
```

Comment:
you may can set latest [hash](https://github.com/pnpm/action-setup/releases/tag/v4.0.0) in here?

Reply:
You might need to specify the commit hash as shown below:

`uses: pnpm/action-setup@fe02b34f77f8bc703788d5817da081398fad5dd2 # v4.0.0`

ref: https://docs.renovatebot.com/modules/manager/github-actions/#additional-information

Reply:
> You might need to specify the commit hash as shown below:
> 
> `uses: pnpm/action-setup@fe02b34f77f8bc703788d5817da081398fad5dd2 # v4.0.0`
> 
> ref: [docs.renovatebot.com/modules/manager/github-actions#additional-information](https://docs.renovatebot.com/modules/manager/github-actions/#additional-information)

Why should I do this?

If you insist on it so much, feel free to add it.

Reply:
Done.